tasks/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -0,0 +1,106 @@
+      script: |
+        #!/usr/bin/env sh
+        #
+        set -e
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        # Set the value for requestResultsFile
+        echo "$(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt" > $(results.requestResultsFile.path)
+
+        # IIB is the default internal pipeline for fbc requests
+        request=$(jq -r '.fbc.request // "iib"' ${DATA_FILE})
+
+        default_build_timeout_seconds="300"
+        iib_overwrite_from_index_credential=$(jq -r \
+          '.fbc.iibOverwriteFromIndexCredential // "iib-overwrite-fromimage-credentials"' ${DATA_FILE})
+        iib_service_config_secret=$(jq -r '.fbc.iibServiceConfigSecret // "iib-services-config"' ${DATA_FILE})
+        iib_service_account_secret=$(jq -r '.fbc.iibServiceAccountSecret // "iib-service-account"' ${DATA_FILE})
+        build_tags=$(jq -r '.fbc.buildTags // "[]"' ${DATA_FILE})
+        add_arches=$(jq -r '.fbc.addArches // "[]"' ${DATA_FILE})
+        build_timeout_seconds=$(jq -r --arg build_timeout_seconds ${default_build_timeout_seconds} \
+            '.fbc.buildTimeoutSeconds // $build_timeout_seconds' ${DATA_FILE})
+        target_index=$(jq -r '.fbc.targetIndex' ${DATA_FILE})
+        fbc_fragment=$(jq -cr '.components[0].containerImage' ${SNAPSHOT_PATH} |tr -d "\n")
+
+        # The internal-request script will create the InternalRequest and wait until it finishes to get its status
+        # If it fails (Failed, Rejected or Timed out) the script will exit and display the reason.
+        echo "Creating InternalRequest to add FBC contribution to index image:"
+        internal-request -r "${request}" \
+            -p binaryImage=$(params.binaryImage) \
+            -p fromIndex=$(params.fromIndex) \
+            -p targetIndex=${target_index} \
+            -p fbcFragment=${fbc_fragment} \
+            -p iibServiceConfigSecret=${iib_service_config_secret} \
+            -p iibServiceAccountSecret=${iib_service_account_secret} \
+            -p iibOverwriteFromIndexCredential=${iib_overwrite_from_index_credential} \
+            -p buildTimeoutSeconds=${build_timeout_seconds} \
+            -p buildTags=${build_tags} \
+            -p addArches=${add_arches} \
+            -t $(params.requestTimeout) |tee $(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt
+
+        internalRequest=$(awk 'NR==1{ print $2 }' $(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt | xargs)
+        echo "done (${internalRequest})"
+
+        # Fetching InternalRequest status and populating results
+        kubectl get internalrequest ${internalRequest} -o jsonpath='{.status.results}' \
+        | tee $(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt
```

Comment:
Question: Why does the result file use txt extension if it's a json?

Reply:
This file is actually unnecessary. I had a previous logic that needed it but now it is just garbage. Removed.

Reply:
Even better :)